include/nbl/builtin/hlsl/glsl_compat/core.hlsl

```diff
@@ -15,6 +15,12 @@ namespace hlsl
 namespace glsl
 {
 
+template<typename T>
+NBL_CONSTEXPR_INLINE_FUNC typename enable_if<is_fundamental<T>::value, T>::type mix(T a, T b, bool c)
+{
+    return c ? b : a;
+}
+
```

Comment:
we're gonna have a conflict here with `master_to_blit`

Reply:
also I think as @Fletterio found out, `?` only works on fundamentals, can't be used with structs, so you need a partial spec (via struct) on `is_fundamental_v` and an `if-else` return needs to happen for non fundamentals

Reply:
or better yet i remove it and just use lerp from `intrinsics.hlsl`

Reply:
`mix` should alias `lerp`